src/Widget/Text.php

```diff
@@ -98,36 +69,41 @@ public function placeholder(string $value): self
         return $new;
     }
 
+    public function pattern(string $value): self
+    {
+        $new = clone $this;
+        $new->attributes['pattern'] = $value;
+        return $new;
+    }
+
     /**
-     * The pattern attribute, when specified, is a regular expression that the input's value must match in order for
-     * the value to pass constraint validation. It must be a valid JavaScript regular expression, as used by the
-     * RegExp type.
+     * A Boolean attribute which, if present, means this field cannot be edited by the user.
+     * Its value can, however, still be changed by JavaScript code directly setting the HTMLInputElement.value
+     * property.
      *
-     * @param string $value
+     * @param bool $value
      *
      * @return static
      *
-     * @link https://www.w3.org/TR/2012/WD-html-markup-20120329/input.text.html#input.text.attrs.pattern
+     * @link https://www.w3.org/TR/2012/WD-html-markup-20120329/textarea.html#textarea.attrs.readonly
      */
-    public function pattern(string $value): self
+    public function readonly(bool $value = true): self
```

Comment:
This method could be removed since it duplicates parent exactly.

Reply:
done.